hazelcast/src/main/java/com/hazelcast/internal/util/graph/BronKerboschCliqueFinder.java

```diff
@@ -0,0 +1,178 @@
+/*
+*  Original work Copyright (c) 2005-2020, by Ewgenij Proschak and Contributors.
+ * Modified work Copyright (c) 2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util.graph;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class BronKerboschCliqueFinder<V> {
+
+    private final Graph<V> graph;
+    private final long nanos;
+    private boolean timeLimitReached;
+    private List<Set<V>> maximumCliques;
+
+    /**
+     * Constructor
+     *
+     * @param graph   the input graph; must be simple
+     * @param timeout the maximum time to wait, if zero no timeout
+     * @param unit    the time unit of the timeout argument
+     */
+    public BronKerboschCliqueFinder(Graph<V> graph, long timeout, TimeUnit unit) {
+        this.graph = Objects.requireNonNull(graph, "Graph cannot be null");
+        if (timeout == 0L) {
```

Comment:
timeout zero seems a special value for infinity, if it is possible to eliminate it, code may be clearer.